mlflow/entities/trace.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+import json
+from dataclasses import asdict, dataclass
+
+from mlflow.entities._mlflow_object import _MLflowObject
+from mlflow.entities.trace_data import TraceData
+from mlflow.entities.trace_info import TraceInfo
+
+
+@dataclass
+class Trace(_MLflowObject):
+    """A trace object. (TODO: Add conceptual guide for tracing.)
+
+    Args:
+        trace_info: A lightweight object that contains the metadata of a trace.
+        trace_data: A container object that holds the spans data of a trace.
+    """
+
+    trace_info: TraceInfo
+    trace_data: TraceData
+
+    def to_json(self) -> str:
```

Comment:
Note: `Trace` needs to be JSON serializable for in-notebook UI rendering.

Reply:
Should we add a from_json class method for a factory conversion from the JSON return from the search API?

Reply:
Definitely we will! But I'd stick to YAGNI here until we decide the spec of search API.

Reply:
UPDATE: discussed offline, we already have ongoing tasks require `to_json` and `from_dict`. I will implement it in this PR:)